### DISCUSSION TITLE
How to assign a port when containerize bentoml

### DISCUSSION THREAD

**Discussion by OriAlpha** [bentoml/BentoML#3761] [Q&A]
When i containerize a model by default its assigned to port 3000, but if i want to run on different port its not possible. 
For example: 

`docker run -it --rm -p 3000:3000 model:tag serve --production`
when i curl to this via 3000 it works

But when i run same on different port as 6000 
`docker run -it --rm -p 6000:6000 model:tag serve --production`
curl just fails, so can assign a port while containerize model 

`bentoml containerize model:latest ` some where here

**Comment by aarnphm** [comment]
`containerize` doesn't have the responsibility to set the container port.

What are you running on port 6000?

If you are running a separate ASGI app on such port, you probably need custom dockerfile and expose that PORT via the dockerfile. See https://docs.bentoml.org/en/latest/guides/containerization.html#dockerfile-template

`docker run -p` exposes the ports of that container, so obv if you don't have anything running on 6000 it won't work.

**Reply by OriAlpha** [reply]
6000 port is free, but curl doesnt work 
```
docker run -it --rm -p 6000:6000 model:gcyks3g47o4zyo3f serve --production
2023-04-17T11:46:15+0000 [INFO] [cli] Service loaded from Bento directory: bentoml.Service(tag="model:gcyks3g47o4zyo3f", path="/home/bentoml/bento/")
2023-04-17T11:46:15+0000 [INFO] [cli] Environ for worker 0: set CPU thread count to 64
2023-04-17T11:46:15+0000 [INFO] [cli] Prometheus metrics for HTTP BentoServer from "/home/bentoml/bento" can be accessed at http://localhost:3000/metrics.
2023-04-17T11:46:16+0000 [ERROR] [cli] Exception in callback <bound method Arbiter.manage_watchers of <circus.arbiter.Arbiter object at 0x7f327730de20>>
```

docker ps 
`1b00ef6ba0fb   model:gcyks3g47o4zyo3f   "/home/bentoml/bento…"   About a minute ago   Up About a minute   3000-3001/tcp, 0.0.0.0:6000->6000/tcp, :::6000->6000/tcp   adoring_solomon`

**Comment by OriAlpha** [comment]
i could see the file structure, where the docker file is create when model is converted to bentoml local format. 
https://docs.bentoml.org/en/latest/concepts/bento.html
I use 
```
model = bentoml.bentos.build(service="svc.py:service",include=['svc.py'],
                                     python=dict(requirements_txt="requirements.txt",),
                                     docker=dict(python_version="3.8"))
```
is there a way to modify in bentoml.bentos.build python function
